### ISSUE TITLE
Moving lines within folds closes the fold (7.4 Patches1-1553)

### ISSUE THREAD

**Issue by tstelzer** [vim/vim#687]
**Setup**
- Using [this precompiled 64bit Vim](https://bintray.com/veegee/generic/vim_x64) on Windows 7 64bit
- The following mappings to move lines [taken from vim.wikia](http://vim.wikia.com/wiki/Moving_lines_up_or_down)

```
nnoremap <A-j> :m .+1<CR>==
nnoremap <A-k> :m .-2<CR>==
inoremap <A-j> <Esc>:m .+1<CR>==gi
inoremap <A-k> <Esc>:m .-2<CR>==gi
vnoremap <A-j> :m '>+1<CR>gv=gv
vnoremap <A-k> :m '<-2<CR>gv=gv
```
- A file that uses folds with `foldmethod=marker` and several closed folds
- **Cursor somewhere within opened fold**, with folds around it **closed**

**expected behaviour**
Line under cursor moves one line up or down, indents the line afterwards _(not that relevant to this bug)_

**observed behaviour**
Line under cursor moves one line up or down, **fold under cursor closes**, and whole fold gets indented _(see following screengrabs)_
![Screengrab 1, Line at the beginning of the fold](http://i.imgur.com/64bFmTc.gif)
![Screengrab 2, Line somewhere within the fold](http://i.imgur.com/yjImoCZ.gif)

_Note that I tested this behaviour with `vim -N -u NONE` and also without the intendation part, with the same results, the screengrabs were for presentational purposes on reddit_

**variation**
I only get the intented result by opening all the folds _before_ using the mappings
![Screengrab 3, opening all the folds beforehand](http://i.imgur.com/LKHHlDC.gif)

**Comment by luchr** [comment]
I have no solution for the underlying problem (of `:m`), but an ugly workaround (which I use):

```
nmap <M-k> :let fdm_sav=&fdm\|:set fdm=manual\|:m-2<CR>:let &fdm=fdm_sav<CR>==
nmap <M-j> :let fdm_sav=&fdm\|:set fdm=manual\|:m+<CR>:let &fdm=fdm_sav<CR>==
vmap <M-k> :<C-U>let fdm_sav=&fdm\|:set fdm=manual\|:'<,'>m'<-2<CR>gv=:let &fdm=fdm_sav<CR>gv
vmap <M-j> :<C-U>let fdm_sav=&fdm\|:set fdm=manual\|:'<,'>m'>+<CR>gv=:let &fdm=fdm_sav<CR>gv
```

The "idea" is to save the current foldmethod (`fdm`), to set it to `manual` for the `:m` command and to set it back to the saved value. Yes ... this is not nice.

**Comment by tstelzer** [comment]
Seems to work, thanks for the temp fix.

**Comment by chrisbra** [comment]
fixed by [v8.0.0459](https://github.com/vim/vim/releases/tag/v8.0.0459)
